libbeat/cmd/instance/metrics.go

```diff
@@ -254,3 +260,79 @@ func reportRuntime(_ monitoring.Mode, V monitoring.Visitor) {
 
 	monitoring.ReportInt(V, "goroutines", int64(runtime.NumGoroutine()))
 }
+
+func reportBeatCgroups(_ monitoring.Mode, V monitoring.Visitor) {
+	V.OnRegistryStart()
+	defer V.OnRegistryFinished()
+
+	pid, err := process.GetSelfPid()
+	if err != nil {
+		logp.Err("error getting PID for self process: %v", err)
+		return
+	}
+
+	cgroups, err := cgroup.NewReader("", true)
+	if err != nil {
+		if err == cgroup.ErrCgroupsMissing {
+			logp.Warn("cgroup data collection disabled: %v", err)
+		} else {
+			logp.Err("cgroup data collection disabled: %v", err)
+		}
+		return
+	}
+	selfStats, err := cgroups.GetStatsForProcess(pid)
+	if err != nil {
+		logp.Err("error getting group status: %v", err)
```

Comment:
Shouldn't that be "error getting cgroup stats"? The error message puzzled me initially..

Reply:
Indeed :)
Thanks for pointing that out, will be fixed by https://github.com/elastic/beats/pull/23413